### DISCUSSION TITLE
IP address and Port number vanishes

### DISCUSSION THREAD

**Discussion by ytrewq271828** [ANLAB-KAIST/KENSv3#198] [Project Q&A 2. Transport Layer (TCP)]
Hello.

I was testing my code with TestEnv_Reliable.TestAccept_Backlog1 file.
When testing the handshake, I confirmed that the port number and IP addresses are delivered in a proper way.
However, after resolving the error in the handshake and testing the accept() call,  I printed the port number and IP address of the client.
Regardless of the modification I conduct, the port number and IP address of the client always becomes zero, thus failing the EXPECT_EQs in the test file.

Stuck in this error. Any ideas on the reason of this error would be helpful.
Thanks in advance.

**Comment by Pusnow** [comment]
> the port number and IP address of the client always becomes zero

Very strange. It looks like buffer-over-run memory bug. Try address sanitizer first: https://github.com/ANLAB-KAIST/KENSv3/wiki/Tip:-Debugging-Memory-Bugs-(Only-for-macOS-and-Linux)

Also, which exact test cases and `EXPECT_EQ`s are you failing?

**Reply by ytrewq271828** [reply]
For TestEnv_Reliable.TestAccept_Backlog1, line 84 of testhandshake.cpp always fails. 
It's strange that other EXPECT_EQs between line 73 and 83 do not complain to me at all.

**Reply by Pusnow** [reply]
Have you implemented `getsockname()` correctly?

Would you let us know the value of `addr.sin_port` and `temp_addr.sin_port`?

**Reply by ytrewq271828** [reply]
![image](https://github.com/ANLAB-KAIST/KENSv3/assets/52207623/5834e4ec-4c3a-4e14-a688-2ba970a7a156)

**Comment by livelykitten** [comment]
When you returning blocked accept call, do you fill in addr parameter?
